Documentation/storage-providers.md

```diff
@@ -0,0 +1,89 @@
+---
+title: Storage Providers
+weight: 12050
+indent: true
+---
+
+# Storage Providers
+
+Rook is the home for operators for multiple storage providers. Each of these storage providers
+has specific requirements and each of them is very independent. There is no runtime dependency
+between the storage providers. Development time is where the storage providers benefit one another.
+
+Rook provides a development framework with a goal of enabling storage providers to create
+operators for Kubernetes to manage their storage layer. As the storage provider community
+grows, we expect this framework to grow as common storage constructs are identified
+that will benefit the community. Rook does not aim to replace other frameworks or
+communities, but to fill gaps not filled by other core projects. Here are
+some examples of other frameworks areas that already meet many community needs:
+
+* [Operator SDK](https://sdk.operatorframework.io/): For creating general operators.
+* [Storage SIG](https://github.com/kubernetes/community/tree/master/sig-storage):
+  For K8s storage API included in K8s, CSI drivers, snapshotting, local storage, and much more.
+* [Controller Runtime](https://github.com/kubernetes-sigs/controller-runtime):
+  Set of libraries for building Controllers for K8s.
+
+Rook does not aim to be a general framework for storage, but to provide a
+very specific set of helper to meet the storage provider needs in the Rook project.
+
+## Rook Framework
+
+Rook provides the following framework to assist storage providers in building an operator:
+
+* Common golang packages shared by storage providers are in the main [Rook repo](https://github.com/rook/rook).
+* Common build scripts for building the operator images are in the main
+  [Rook repo](https://github.com/rook/rook/tree/master/build)].
+* Each provider has its own repo under the [Rook org](https://github.com/rook).
+  * Multiple community members are given push access to the repo, including
+    owners of the storage provider, Rook steering committee members,
+    and other Rook maintainers if deemed helpful or necessary by the steering
+    committee. Maintainers for the new provider are added according to the
+    [governance](https://github.com/rook/rook/blob/master/GOVERNANCE.md).
+  * Providers added to Rook prior to 2020 are grandfathered into the main
+    [Rook repo](https://github.com/rook/rook).
+* Storage providers must follow the Rook [governance](https://github.com/rook/rook/blob/master/GOVERNANCE.md)
+  in the interest of the good of the overall project. Storage providers have
+  autonomy in their feature work, while collaboration
+  with the community is expected where there are shared features.
+* A quarterly release cadence is in place for the operators in the main Rook repo.
+  Operators in their own repo define their own cadence and versioning scheme as desired.
+  * Storage providers own their release process, while following Rook best practices to
```

Comment:
perhaps mention here again all the shared build/release scripts and automation, CI infrastructure, artifacts hosting, etc. that the project provides